forc-plugins/forc-node/src/chain_config.rs

```diff
@@ -0,0 +1,579 @@
+    sha: String,
+    download_url: Option<String>,
+    #[serde(rename = "type")]
+    content_type: String,
+}
+
+/// `ConfigFetcher` is responsible for github api integration related to the
+/// configuration operations.
+/// Basically checks remote hash of the corresponding chain configuration.
+/// If there is a mismatch between local and remote instance, overrides the
+/// local instance with remote changes for testnet and mainnet configurations.
+///
+/// For local chain configuration, we only check for existence of it locally.
+/// If the local chain configuration is missing in user's local,
+/// `ConfigFetcher` fetches it but remote updates are not tracked for it.
+pub struct ConfigFetcher {
+    client: reqwest::Client,
+    base_url: String,
+    config_vault: PathBuf,
+}
+
+impl Default for ConfigFetcher {
+    /// Creates a new fetcher to interact with github.
+    /// By default user's chain configuration vault is at: `~/.forc/chainspecs`
+    fn default() -> Self {
+        Self {
+            client: reqwest::Client::new(),
+            base_url: "https://api.github.com".to_string(),
+            config_vault: user_forc_directory().join(CONFIG_FOLDER),
+        }
+    }
+}
+
+impl ConfigFetcher {
+    #[cfg(test)]
+    /// Override the base url, to be used in tests.
+    pub fn with_base_url(base_url: String) -> Self {
+        Self {
+            client: reqwest::Client::new(),
+            base_url,
+            config_vault: user_forc_directory().join(CONFIG_FOLDER),
+        }
+    }
+
+    #[cfg(test)]
+    pub fn with_test_config(base_url: String, config_vault: PathBuf) -> Self {
+        Self {
+            client: reqwest::Client::new(),
+            base_url,
+            config_vault,
+        }
+    }
+
+    fn get_base_url(&self) -> &str {
+        &self.base_url
+    }
+
+    fn build_api_endpoint(&self, folder_name: &str) -> String {
+        format!(
+            "{}/repos/FuelLabs/{}/contents/{}",
+            self.get_base_url(),
+            CHAIN_CONFIG_REPO_NAME,
+            folder_name,
+        )
+    }
+
+    /// Fetches contents from github to get hashes and download urls for
+    /// contents of the remote configuration repo at:
+    /// https://github.com/FuelLabs/chain-configuration/
+    async fn check_github_files(
+        &self,
+        conf: &ChainConfig,
+    ) -> anyhow::Result<Vec<GithubContentDetails>> {
+        let folder_name = match conf {
+            ChainConfig::Local => LOCAL_CONFIG_FOLDER_NAME,
+            ChainConfig::Testnet => TESTNET_CONFIG_FOLDER_NAME,
+            ChainConfig::Ignition => IGNITION_CONFIG_FOLDER_NAME,
+        };
+        let api_endpoint = self.build_api_endpoint(folder_name);
+
+        let response = self
+            .client
+            .get(&api_endpoint)
+            .header("User-Agent", "forc-node")
+            .send()
+            .await?;
+
+        if !response.status().is_success() {
+            bail!("failed to fetch updates from github")
```

Comment:
nit: could provide details on why this failed (e.g. could know at glance if gh is down or something):

```suggestion
            bail!("Failed to fetch updates from GitHub: status {}, body: {}", 
                response.status(),
                response.text().await?)
```

Reply:
I'll add this later on not to dismiss reviews